lib/puppet/application/master.rb

```diff
@@ -186,10 +186,11 @@ def main
 
     if Puppet.features.root?
       begin
-        Puppet::Util.chuser
+        if Etc.getpwnam(Puppet[:user])
+          Puppet::Util.chuser
+        end
       rescue => detail
-        Puppet.log_exception(detail, "Could not change user to #{Puppet[:user]}: #{detail}")
-        exit(39)
+        Puppet.warning "Could not change user to #{Puppet[:user]}: #{detail}. Continuing as root."
```

Comment:
@joshcooper right, that was my mistake. If we use `Puppet::Type.type(:user)...` like you mentioned above to check, we probably won't have to deal with an extra level of exceptions, so I'll try that.